ml/src/test/scala/frameless/ml/TypedEncoderInstancesTests.scala

```diff
@@ -0,0 +1,60 @@
+package frameless
+package ml
+
+import frameless.TypedDataset
+import org.scalacheck.Prop._
+import org.apache.spark.ml.linalg._
+import org.apache.spark.ml.regression.DecisionTreeRegressor
+import Generators._
+
+import scala.util.Random
+
+class TypedEncoderInstancesTests extends FramelessMlSuite {
+
+  test("Vector encoding is injective using collect()") {
+    val prop = forAll { vector: Vector =>
+      TypedDataset.create(Seq(vector)).collect().run() ?= Seq(vector)
+    }
+    check(prop)
+  }
+
+  test("Matrix encoding is injective using collect()") {
+    val prop = forAll { matrix: Matrix =>
+      TypedDataset.create(Seq(matrix)).collect().run() ?= Seq(matrix)
+    }
+    check(prop)
+  }
+
+  test("Vector is encoded as VectorUDT and thus can be run in a Spark ML model") {
+    case class Input(features: Vector, label: Double)
+
+    val prop = forAll { trainingData: Matrix =>
+      // Spark 2.1.x decision tree implementation has a bug with a dataset with constant label,
+      // so we simulate two different labels
+      // See https://issues.apache.org/jira/browse/SPARK-18036
+      val inputs = trainingData.rowIter.toVector.zipWithIndex.map {
+        case (vector, i) => Input(vector, (i % 2).toDouble)
+      }
+
+      val inputsDS = TypedDataset.create(inputs)
```

Comment:
What is the return type here? for inputsDS

Reply:
ping

Reply:
As I see from the code it's `TypedDataset[Input]`

Reply:
Yes it's indeed ```TypedDataset[Input]```